dashboard/src/utils/axiosInstance.js

```diff
@@ -0,0 +1,41 @@
+import axios from "axios";
+import { LOCAL_BASE_URL } from "./apiConstants";
+import Cookies from "js-cookie";
+import { showSessionExpired } from "actions/toastActions";
+import store from "store/store";
+
+const { dispatch } = store;
+
+let token = Cookies.get("token") ? Cookies.get("token") : null;
+
+const axiosInstance = axios.create({
+  baseURL: LOCAL_BASE_URL,
+  responseType: "json",
+  headers: {
+    Authorization: `Bearer ${token}`,
+  },
+});
+
+axiosInstance.interceptors.request.use(async (req) => {
+  token = Cookies.get("token") ? Cookies.get("token") : null;
+  req.headers.Authorization = `Bearer ${token}`;
+
+  return req;
+});
+/** Intercept any unauthorized request.
+ * dispatch logout action accordingly **/
+const UNAUTHORIZED = 401;
+axiosInstance.interceptors.response.use(
+  (response) => response,
+  (error) => {
+    if (error?.response) {
+      const { status } = error.response;
+      if (status === UNAUTHORIZED) {
+        dispatch(showSessionExpired());
+      }
```

Comment:
What do/should we do if the status is something other than `UNAUTHORIZED`?  Really nothing?

Reply:
Promise is rejected with error and it gets handled in the corresponding functions `catch` block